query-engine/request-handlers/src/load_executor.rs

```diff
@@ -70,6 +70,15 @@ async fn mysql(
     features: PreviewFeatures,
 ) -> query_core::Result<Box<dyn QueryExecutor + Send + Sync>> {
     trace!("Loading MySQL query connector...");
+
+    if source.provider == "@prisma/mysql" {
+        // TODO: change this to a new PrismaMysql connector once
+        // https://github.com/prisma/client-planning/issues/326 is ready
+        let mysql = Mysql::from_source(source, url, features).await?;
+        trace!("Loaded @prisma/mysql query connector.");
+        return Ok(sql_executor(mysql, false));
+    }
+
     let mysql = Mysql::from_source(source, url, features).await?;
     trace!("Loaded MySQL query connector.");
     Ok(sql_executor(mysql, false))
```

Comment:
Note: this won't be a pattern match because the `PrismaMysql` connector will be a separate struct, thus `match` would not be available due to mismatching concrete types.

Reply:
In case we want to add conditionals inside the `Mysql` connector itself, we can revert https://github.com/prisma/prisma-engines/pull/3972/commits/5e5568661907be4a8ef8d29ff3522a12b96a4e3e and merge right away.

Reply:
We can decide that later, when we actually plug in the node driver impl into the mysql connector.